src/custom-element.js

```diff
@@ -1434,10 +1447,16 @@ function createBaseAmpElementProto(win) {
  */
 export function registerElement(win, name, implementationClass) {
   knownElements[name] = implementationClass;
+  const klass = createBaseCustomElementClass(win, name);
 
-  win.document.registerElement(name, {
-    prototype: createAmpElementProto(win, name),
-  });
+  const supportsCustomElementsV1 = 'customElements' in win;
+  if (supportsCustomElementsV1) {
+    win['customElements'].define(name, klass);
```

Comment:
Closure type checker complains that `customElements` is not defined on win, so I think this would prevent Closure from renaming the property. Not sure if there's a better way to fix this.